### PR TITLE
Fix MakeLocalFunctionStatic for top-level local function

### PR DESCRIPTION
**Pull request by jcouv**
Fixes https://github.com/dotnet/roslyn/issues/53179

### CODE REVIEW

File: src/Analyzers/CSharp/Tests/MakeLocalFunctionStatic/MakeLocalFunctionStaticTests.cs [hunk 1]
```diff
@@ -510,5 +510,22 @@ void B()
...
+void [||]A()
+{
+}", @"
+static void A()
+{
+}",
+parseOptions: CSharp8ParseOptions);
+
+            //await TestAsync(initialMarkup, expectedMarkup, WithRegularOptions(parameters));
+
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/Tests/MakeLocalFunctionStatic/MakeLocalFunctionStaticTests.cs:528]:
```suggestion
```

File: src/Features/CSharp/Portable/MakeLocalFunctionStatic/MakeLocalFunctionStaticCodeFixProvider.cs [hunk 2]
```diff
@@ -54,6 +55,11 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
             }
 
             return Task.CompletedTask;
+
+            static SyntaxNode unwrapGlobalStatement(SyntaxNode node)
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/MakeLocalFunctionStatic/MakeLocalFunctionStaticCodeFixProvider.cs:59]:
```suggestion
            static SyntaxNode UnwrapGlobalStatement(SyntaxNode node)
```

File: src/Features/CSharp/Portable/MakeLocalFunctionStatic/MakeLocalFunctionStaticCodeFixProvider.cs [hunk 3]
```diff
@@ -45,7 +46,7 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
             Document document, ImmutableArray<Diagnostic> diagnostics,
             SyntaxEditor editor, CancellationToken cancellationToken)
         {
-            var localFunctions = diagnostics.SelectAsArray(d => d.AdditionalLocations[0].FindNode(cancellationToken));
+            var localFunctions = diagnostics.SelectAsArray(d => unwrapGlobalStatement(d.AdditionalLocations[0].FindNode(cancellationToken)));
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/MakeLocalFunctionStatic/MakeLocalFunctionStaticCodeFixProvider.cs:49]:
```suggestion
            var localFunctions = diagnostics.SelectAsArray(d => d.AdditionalLocations[0].FindNode(getInnermostNodeForTie: true, cancellationToken));
```

File: src/Features/CSharp/Portable/MakeLocalFunctionStatic/MakeLocalFunctionStaticCodeFixProvider.cs [hunk 4]
```diff
@@ -54,6 +55,11 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
             }
 
             return Task.CompletedTask;
+
+            static SyntaxNode unwrapGlobalStatement(SyntaxNode node)
+            {
+                return node is GlobalStatementSyntax globalStatement ? globalStatement.Statement : node;
+            }
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/MakeLocalFunctionStatic/MakeLocalFunctionStaticCodeFixProvider.cs:62]:
```suggestion
```

File: src/Analyzers/CSharp/Tests/MakeLocalFunctionStatic/MakeLocalFunctionStaticTests.cs (hunk 1 above)
Comment (jcouv) [reply src/Analyzers/CSharp/Tests/MakeLocalFunctionStatic/MakeLocalFunctionStaticTests.cs:528]: Oops, sorry about that

File: src/Analyzers/CSharp/Tests/MakeLocalFunctionStatic/MakeLocalFunctionStaticTests.cs [hunk 5]
```diff
@@ -508,6 +508,20 @@ void B()
         }
     }
 }",
+parseOptions: CSharp8ParseOptions);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeLocalFunctionStatic)]
+        [WorkItem(53179, "https://github.com/dotnet/roslyn/issues/53179")]
```
Comment (Youssef1313) [inline src/Analyzers/CSharp/Tests/MakeLocalFunctionStatic/MakeLocalFunctionStaticTests.cs:515]: This issue is about about the [`CodeRefactoringProvider`](https://github.com/dotnet/roslyn/blob/main/src/Features/CSharp/Portable/MakeLocalFunctionStatic/MakeLocalFunctionStaticCodeRefactoringProvider.cs), not the [`CodeFixProvider`](https://github.com/dotnet/roslyn/blob/main/src/Features/CSharp/Portable/MakeLocalFunctionStatic/MakeLocalFunctionStaticCodeFixProvider.cs).

Might be good to add a [test](https://github.com/dotnet/roslyn/blob/main/src/EditorFeatures/CSharpTest/MakeLocalFunctionStatic/MakeLocalFunctionStaticRefactoringTests.cs) to the coderefactoring if it already works.

Comment (jcouv) [reply src/Analyzers/CSharp/Tests/MakeLocalFunctionStatic/MakeLocalFunctionStaticTests.cs:515]: Thanks, added a test. They do share the code so worked fine :-)

File: src/Features/CSharp/Portable/MakeLocalFunctionStatic/MakeLocalFunctionStaticCodeFixProvider.cs [hunk 6]
```diff
@@ -12,6 +12,7 @@
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.CodeActions;
 using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/MakeLocalFunctionStatic/MakeLocalFunctionStaticCodeFixProvider.cs:15]: Needed?

**Review by CyrusNajmabadi — APPROVED** [review]

**Pull request merged**